code-challenges.test.js

```diff
@@ -14,39 +14,167 @@
+    //Tests:       0 total
+    //Snapshots:   0 total
+    //Time:        0.521 s
+
 
 // b) Create the function that makes the test pass.
 
+// pseudocode 
+// declare a funtion codedword
+//itirate through the array and find the strings that match "a, e, i, o,u"
+// make an object to make a table to map out the characters to their corresponding values
+// the coded message converts "a" to 4, "e" to 3, "i" to 1, and "o" to 0.
+// we use the replace method 
+// use .toLowercase to catch letters that are capitilized
+// we use charCodeMap[char.toLowercase()] to retrieve the corresponding matched character from the object
+// we need to return the orinal character for the characters that arent in our map
+// well use the OR '||' operator for the last step 
+// the output will be our encoded message 
+
+const secretCode = (message) => {
+    const charCodeMap = { 'a': '4', 'e': '3', 'i': '1', 'o': '0' }
+    return message.replace(/[aeio]/gi, char => charCodeMap[char.toLowerCase()] || char)
+}
+
+// Test Suites: 1 passed, 1 total
+// Tests:       3 passed, 3 total
+// Snapshots:   0 total
+// Time:        0.306 s
+// Ran all test suites.
+// ✨  Done in 1.12s.
+// s
+
 // --------------------2) Create a function that takes in an array of words and a single letter and returns an array of all the words containing that particular letter.
 
 // a) Create a test with expects statement using the variable provided.
 
-const fruitArray = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach", "Kiwi"]
+describe("wordsWithLetter", () => {
+    const fruitArray = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach", "Kiwi"]
+
+    it("should return an array of words containing 'a'", () => {
+        const letterA = "a"
+        expect(wordsWithLetter(fruitArray, letterA)).toEqual(["Mango", "Apricot", "Peach"])
+    })// Expected output: ["Mango", "Apricot", "Peach"]
+    it("should return an array of words containing 'e'", () => {
+        const letterE = "e"
+        expect(wordsWithLetter(fruitArray, letterE)).toEqual(["Cherry", "Blueberry", "Peach"])// Expected output: ["Cherry", "Blueberry", "Peach"]
+    })
+
+})
+
+
 
-const letterA = "a"
-// Expected output: ["Mango", "Apricot", "Peach"]
-const letterE = "e"
-// Expected output: ["Cherry", "Blueberry", "Peach"]
 
 // b) Create the function that makes the test pass.
+// Psuedo code
+// declare a function
+// use the words array and the letter as your paramaters
+// iterate through the array checking for the string that matches your string
+// if it matches return that word in youre new array 
+// declare a new function thatll be filtered words
+// use .filter method to make a new array 
+// use .tolowercase to include capital letters
+// use .includes to check if the word has the specified letter
+// return filteredWords
+
+const wordsWithLetter = (wordsArray, letter) => {
+    const filteredWords = wordsArray.filter(word => word.toLowerCase().includes(letter.toLowerCase()))
+    return filteredWords 
+}
```

Comment:
Looks good, but no need to make `filteredWords` at all, you can just return the data itself.